core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ShaclSailConnection.java

```diff
@@ -938,6 +946,16 @@ void applyTransactionSettings(Settings transactionSettingsLocal) {
 				validationApproach = transactionSettingsLocal.validationApproach;
 			}
 
+			assert transactionSettingsLocal.isolationLevel == null;
```

Comment:
I'm not sure I follow the logic of this assertion. Why are you assuming that the isolationLevel in the supplied `Settings` object will always be `null`?

Reply:
It's because we don't support overriding the isolation level.